contracts/eth/nearbridge/contracts/NearDecoder.sol

```diff
@@ -30,12 +33,20 @@ library NearDecoder {
     struct BlockProducer {
         PublicKey publicKey;
         uint128 stake;
+        // Flag indicating if this validator proposed to be a chunk-only producer (i.e. cannot become a block producer).
+        bool isChunkOnly;
     }
 
     function decodeBlockProducer(Borsh.Data memory data) internal pure returns (BlockProducer memory res) {
+        uint8 validator_version = data.decodeU8();
         data.skipBytes();
         res.publicKey = data.decodePublicKey();
         res.stake = data.decodeU128();
+        if (validator_version == VALIDATOR_V2) {
+            res.isChunkOnly = data.decodeU8() != 0;
```

Comment:
Can't you use `validator_version` here instead?

Reply:
We need to extract it from borsh-data (changing the cursor to be precise). Otherwise, we will have some data left.